arbnode/espresso_caff_node.go

```diff
@@ -282,16 +357,25 @@ func (n *EspressoCaffNode) nextMessage() (*espressostreamer.MessageWithMetadataA
 		return nil, nil
 	}
 
-	if messageWithMetadataAndPos.MessageWithMeta.DelayedMessagesRead == n.delayedCount+1 {
+	if n.nextDelayedCount > 0 && messageWithMetadataAndPos.MessageWithMeta.DelayedMessagesRead == n.nextDelayedCount {
```

Comment:
Not sure why you rename `delayedCount` to `nextDelayedCount`. delayed count indicates number of delayed messages that have been read.

It should always start from 1, which means that initial message has been read. So `n.nextDelayedCount>0` is unnecessary.

Also, the `MessageWithMeta.DelayedMessagesRead` starts from 1 and if it is greater than caff node's delayed count, means this message is a delayed message. So I think this change is not correct.

Reply:
Yeah I see how I misinterpreted that I think. if we don't want to remove the +1 on n.delayed I think we actually should be starting `n.delayedCount` at 0 if we don't want to exclude the delayed message with index 1.